### DISCUSSION TITLE
Always re-rendering?

### DISCUSSION THREAD

**Discussion by bradherman** [jaredh159/tailwind-react-native-classnames#240] [Q&A] [answered]
After installing `why-did-you-render`, I started noticing all of my components always re-render when going through the hierarchy and it's because the style prop is never the same... Doing a quick check I noticed that:

```
tw`h-full` === tw`h-full` => false
```

Does using twrnc cause all components to always re-render unless the style is wrapped up in a memo?  Or am I missing something?

Changing to this:

```
const wrapperStyle = useMemo(() => tw`h-full`, [])

<View style={wrapperStyle}>
```

eliminates extra renders, but seems like a lot of extra overhead

**Comment by jaredh159** [comment, marked as answer]
good question. i've not had any perf issues from this myself, but it didn't seem right that two identical calls to `tw` would not be `===` so i poked around a bit and found i was spreading in one spot where i didn't need to, thus creating a new object which would trigger a rerender.

here's the PR if you're interested:

https://github.com/jaredh159/tailwind-react-native-classnames/pull/241

i've already merged it and cut a new release: `3.6.4`. would you mind testing it to see if it seems better?  there are still cases where you'll fail the `===` test, specifically if you're also passing in a style object to be merged, but that is far rarer than just passing a string of utilities.

**Comment by bradherman** [comment]
Awesome!  I'll get it updated in my proj and try it out
